pkg/server/events.go

```diff
@@ -236,9 +267,85 @@ func handleSandboxExit(e *eventtypes.TaskExit, sb sandboxstore.Sandbox) {
 	})
 	if err != nil {
 		logrus.WithError(err).Errorf("Failed to update sandbox %q state", e.ContainerID)
-		// TODO(random-liu): [P0] Enqueue the event and retry.
-		return
+		return err
 	}
 	// Using channel to propagate the information of sandbox stop
 	sb.Stop()
+	return nil
+}
+
+func (em *eventMonitor) isBackOffIng(evt interface{}) (string, bool) {
+	key := ""
+	switch evt.(type) {
+	case *eventtypes.TaskExit:
+		key = evt.(*eventtypes.TaskExit).ContainerID
+	case *eventtypes.TaskOOM:
+		key = evt.(*eventtypes.TaskOOM).ContainerID
+	default:
+		return key, false
+	}
+	return key, len(em.backOffQueue[key].events) != 0
+}
+
+// enBackOff start to backOff and put event to the tail of queue
+func (em *eventMonitor) enBackOff(key string, evt interface{}) {
+	queue := em.backOffQueue[key]
+	queue.events = append(queue.events, evt)
+	if len(queue.events) == 1 {
+		queue.timer = em.newBackOffTimer(key)
+	}
+	em.backOffQueue[key] = queue
+}
+
+// enBackOff get out one event from the begin of queue
+func (em *eventMonitor) deBackOff(key string) interface{} {
+	queue := em.backOffQueue[key]
+	if len(queue.events) == 0 {
+		return nil
+	}
+	evt := queue.events[0]
+	queue.timer.Stop()
+	queue.events = queue.events[1:]
+	if len(queue.events) != 0 {
+		queue.timer = em.newBackOffTimer(key)
+	}
+	em.backOffQueue[key] = queue
+	return evt
+}
+
+// enBackOff start to backOff again and put event to the begin of queue
+func (em *eventMonitor) reBackOff(key string, evt interface{}) {
+	newEvents := []interface{}{evt}
+	queue := em.backOffQueue[key]
```

Comment:
Just requeue, and always stop timer as long as it is not `nil`.

No need to check length.

Reply:
If then, We will stop timer twice when there is only one event.